### PR TITLE
Fixing issue where a cancelled job would not notify pipe is was finished

### PR DESCRIPTION
**Pull request by thechrisjohnson**

### CODE REVIEW

File: src/Agent.Listener/JobDispatcher.cs [hunk 1]
```diff
@@ -265,6 +265,7 @@ public bool Cancel(JobCancelMessage jobCancelMessage)
 
                 if (jobRequestCancellationToken.IsCancellationRequested)
                 {
+                    notification.JobFinished(message.JobId);
```
Comment (ericsciple) [inline src/Agent.Listener/JobDispatcher.cs]: The others are on the line after `CompleteJobRequestAsync`, does it matter? `CompleteJobRequestAsync` is a private method, would it be better to move the call into there?

**Comment by bryanmacfarlane** [comment]
preview about to drop.  If this is important for hosted, let's get it in

**Comment by ericsciple** [comment]
looks good

**Comment by bryanmacfarlane** [comment]
CJ - I'm merging to get maximum coverage as devs run in last few days before release.

**Pull request merged**